### ISSUE TITLE
dials.image_viewer fails to open experiments with PolychromaticBeam objects

### ISSUE THREAD

**Issue by PrinceWalnut** [dials/dials#2814]
Running dials.image_viewer on some files I have which contain a `PolychromaticBeam` object, and the viewer fails to open because of the lack of an `s0` vector. Traceback below. 

```(base) [rahewitt@holy7c06105 examples]$ dials.image_viewer optimized.*
DIALS (2018) Acta Cryst. D74, 85-97. https://doi.org/10.1107/S2059798317017235
The following parameters have been modified:

input {
  experiments = optimized.expt
  reflections = optimized.refl
}

Traceback (most recent call last):
  File "/n/holyscratch01/hekstra_lab/rahewitt/test_pinkdials/dials/conda_base/bin/dials.image_viewer", line 8, in <module>
    sys.exit(run())
             ^^^^^
  File "/n/holyscratch01/hekstra_lab/rahewitt/test_pinkdials/dials/conda_base/lib/python3.11/contextlib.py", line 81, in inner
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "/n/holyscratch01/hekstra_lab/rahewitt/test_pinkdials/dials/modules/dials/src/dials/command_line/image_viewer.py", line 209, in run
    show_image_viewer(params=params, reflections=reflections, experiments=experiments)
  File "/n/holyscratch01/hekstra_lab/rahewitt/test_pinkdials/dials/modules/dials/src/dials/command_line/image_viewer.py", line 174, in show_image_viewer
    wrapper.display(experiments=experiments, reflections=reflections)
  File "/n/holyscratch01/hekstra_lab/rahewitt/test_pinkdials/dials/modules/dials/src/dials/util/image_viewer/spotfinder_wrap.py", line 86, in display
    self.frame.load_image(chooser_wrapper(imagesets[0], 0))
  File "/n/holyscratch01/hekstra_lab/rahewitt/test_pinkdials/dials/modules/dials/src/dials/util/image_viewer/spotfinder_frame.py", line 753, in load_image
    super().load_image(
  File "/n/holyscratch01/hekstra_lab/rahewitt/test_pinkdials/dials/modules/dials/src/dials/util/image_viewer/slip_viewer/frame.py", line 465, in load_image
    panel_id, beam_pixel_fast, beam_pixel_slow = self.get_beam_center_px()
                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/n/holyscratch01/hekstra_lab/rahewitt/test_pinkdials/dials/modules/dials/src/dials/util/image_viewer/slip_viewer/frame.py", line 371, in get_beam_center_px
    raise e
  File "/n/holyscratch01/hekstra_lab/rahewitt/test_pinkdials/dials/modules/dials/src/dials/util/image_viewer/slip_viewer/frame.py", line 367, in get_beam_center_px
    panel_id, (x_mm, y_mm) = detector.get_ray_intersection(beam.get_s0())
                                                           ^^^^^^^^^^^^^
RuntimeError: Please report this error at https://github.com/dials/dials/issues: dxtbx Error: /n/holyscratch01/hekstra_lab/rahewitt/test_pinkdials/dials/modules/dxtbx/src/dxtbx/../dxtbx/
model/beam.h(741): PolychromaticBeam has no fixed s0
```

**Comment by toastisme** [comment]
Ah sorry this should be fixed with #2781

**Comment by dagewa** [comment]
Assume this is fixed (please re-open if not)
